src/vhost_user/master.rs

```diff
@@ -452,6 +458,43 @@ impl VhostUserMaster for Master {
         Ok(())
     }
 
+    fn get_inflight_fd(&mut self) -> Result<(VhostUserInflight, RawFd)> {
+        let mut node = self.node();
+        if node.acked_protocol_features & VhostUserProtocolFeatures::INFLIGHT_SHMFD.bits() == 0 {
+            return error_code(VhostUserError::InvalidOperation);
+        }
+
+        let hdr = node.send_request_header(MasterReq::GET_INFLIGHT_FD, None)?;
+        let (inflight, fds) = node.recv_reply_with_fds::<VhostUserInflight>(&hdr)?;
+
+        if let Some(fds) = fds {
+            if fds.len() == 1 {
+                return Ok((inflight, fds[0]));
+            }
```

Comment:
Should close the received fds to avoid fd leakage.

Reply:
Yes I'll fix that